### DISCUSSION TITLE
try to avoid render auth screen because of the behaviour from onAuthStateChange

### DISCUSSION THREAD

**Discussion by megacherry** [supabase/supabase#2328] [Questions]
hi folks,

i use react native and react navigation. I switch my screens between my AppStack and AuthStack. Based on this answer i implemented onAuthStateChange in that way: (https://github.com/supabase/gotrue-js/issues/23#issuecomment-743032138)

Because of the way how onAuthStateChange is working i get the existing session after the second render. And i will see my AuthScreen milliseconds after it switch to my AppStack.

`return (
    <NavigationContainer>
      {user ? <AppStack /> : <AuthStack />}
    </NavigationContainer>
  );`

Can somebody help me out? :-)

Andreas

**Comment by megacherry** [comment]
does nobody else have this issue? :-(

**Comment by w3b6x9** [comment]
@megacherry i think i had a similar issue and solved it like this: https://github.com/w3b6x9/supabeeper/blob/main/src/navigation/index.jsx. Feel free to check out my code to see if it fixes your issue!
